qiskit/transpiler/target.py

```diff
@@ -484,6 +484,29 @@ def operations_for_qargs(self, qargs):
             raise KeyError(f"{qargs} not in target.")
         return [self._gate_name_map[x] for x in self._qarg_gate_map[qargs]]
 
+    def instruction_supported(self, operation_name, qargs):
+        """Return whether the instruction (operation + qubits) is supported by the target
+
+        Args:
+            operation_name (str): The name of the operation for the instruction
+            qargs (tuple): The tuple of qubit indices for the instruction
+
+        Returns:
+            bool: Returns ``True`` if the instruction is supported and ``False`` if it isn't.
+
+        """
+        return_value = False
+        if operation_name in self._gate_map:
+            if qargs in self._gate_map[operation_name]:
```

Comment:
The type hint says `qargs` must be a `tuple`, and this will fail loudly if it's a list (which is good).  Is there any chance that callers may be programmatically generating `qargs` as lists? If so, we could normalise `qargs = tuple(qargs)` at the start.  That's a no-op if `qargs` already is a `tuple`, so it shouldn't affect performance in the preferred case.

It's ok to deliberately omit it if we want to _force_ people to pass tuples, though.  (I'm fine with that.)

Reply:
It's easy enough to add, it's probably good safety because qargs in a circuit and dagcircuit node are almost always lists and it's just a nice thing to have in case people make a mistake

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7778/commits/cb11e30563ecd7e7ba80437c96fca6eeddbb2210